com.unity.netcode.gameobjects/Runtime/Spawning/INetworkPrefabInstanceHandlerWithData.cs

```diff
@@ -0,0 +1,34 @@
+using System;
+using System.Collections.Generic;
+using Unity.Collections;
+using UnityEngine;
+
+namespace Unity.Netcode
+{
+    /// <summary>
+    /// Specialized version of <see cref="INetworkPrefabInstanceHandler"/> that receives
+    /// custom instantiation data injected by the server before spawning.
+    /// </summary>
+    public abstract class NetworkPrefabInstanceHandlerWithData<T> : INetworkPrefabInstanceHandlerWithData where T : struct, INetworkSerializable
+    {
+        public abstract NetworkObject Instantiate(ulong ownerClientId, Vector3 position, Quaternion rotation, T instantiationData);
+        
+        public abstract void Destroy(NetworkObject networkObject);
+
+        bool INetworkPrefabInstanceHandlerWithData.HandlesDataType<U>() => typeof(T) == typeof(U);
+       
+        NetworkObject INetworkPrefabInstanceHandlerWithData.Instantiate(ulong ownerClientId, Vector3 position, Quaternion rotation, FastBufferReader reader)
+        {
+            reader.ReadValueSafe(out T _payload);
+            return Instantiate(ownerClientId, position, rotation, _payload);
```

Comment:
I think we need to re-assign the byte array to the created `NetworkObject` at this point. Distributed authority means that at any point the client might become the authority on the object.